newm-chain/src/main/kotlin/io/newm/chain/daemon/BlockDaemon.kt

```diff
@@ -561,10 +582,150 @@ class BlockDaemon(
+                    val blockDelayQueue = LinkedList<Block>()
+                    blockFlow.collect { block ->
+                        // Prune any rolled back blocks
+                        blockDelayQueue.removeIf { bdqBlock -> bdqBlock.header.blockHeight >= block.header.blockHeight }
+                        // Add this new block to our queue
+                        blockDelayQueue.add(block)
+                        // Wait for 3 confirmations before processing
+                        if (blockDelayQueue.size > 3) {
+                            val oldestBlock = blockDelayQueue.poll()
+                            val spentUtxoMap = oldestBlock.toSpentUtxoMap()
+                            val createdUtxoMap = oldestBlock.toCreatedUtxoMap()
+                            val transactionIds = spentUtxoMap.keys + createdUtxoMap.keys
+                            val monitorAddressResponsesMap = mutableMapOf<String, MutableList<MonitorAddressResponse>>()
+
+                            transactionIds.forEach { transactionId ->
+                                monitorAddresses.forEach { monitorAddress ->
+                                    val spentAddressUtxos: List<Utxo> =
+                                        spentUtxoMap[transactionId]?.mapNotNull { spentUtxo ->
+                                            ledgerRepository.queryUtxoHavingAddress(
+                                                monitorAddress,
+                                                spentUtxo.hash,
+                                                spentUtxo.ix.toInt()
+                                            )
+                                        }?.map { utxo ->
+                                            Utxo.newBuilder().apply {
+                                                hash = utxo.hash
+                                                ix = utxo.ix
+                                                lovelace = utxo.lovelace.toString()
+                                                utxo.datumHash?.let { datumHash = it }
+                                                utxo.datum?.let { datum = it }
+                                                utxo.nativeAssets.forEach { nativeAsset ->
+                                                    addNativeAssets(
+                                                        NativeAsset.newBuilder().apply {
+                                                            policy = nativeAsset.policy
+                                                            name = nativeAsset.name
+                                                            amount = nativeAsset.amount.toString()
+                                                        }
+                                                    )
+                                                }
+                                            }.build()
+                                        } ?: emptyList()
```

Comment:
May you could use `.orEmpty() ` here.

Reply:
Fixed it here and a few other places in the code. Always learning idiomatic ways to do things. :+1: